examples/class-components/2-Split-Card.js

```diff
@@ -0,0 +1,155 @@
+// @noflow
+
+import React from 'react';
+import {loadStripe} from '@stripe/stripe-js';
+import {
+  CardNumberElement,
+  CardCvcElement,
+  CardExpiryElement,
+  Elements,
+  ElementsConsumer,
+} from '../../src';
+
+import {logEvent, Result, ErrorResult} from '../util';
+import '../styles/common.css';
+
+const ELEMENT_OPTIONS = {
+  style: {
+    base: {
+      fontSize: '18px',
+      color: '#424770',
+      letterSpacing: '0.025em',
+      '::placeholder': {
+        color: '#aab7c4',
+      },
+    },
+    invalid: {
+      color: '#9e2146',
+    },
+  },
+};
+
+class CheckoutForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {name: '', postal: '', result: null};
+  }
+
+  handleSubmit = async (event) => {
+    event.preventDefault();
+    const {stripe, elements} = this.props;
+    const {name, postal} = this.state;
+
+    if (!stripe || !elements) {
+      // Stripe.js has not loaded yet. Make sure to disable
+      // form submission until Stripe.js has loaded.
+      return;
+    }
+
+    const cardElement = elements.getElement(CardNumberElement);
+
+    const payload = await stripe.createPaymentMethod({
+      type: 'card',
+      card: cardElement,
+      billing_details: {
+        name,
+        address: {
+          postal_code: postal,
+        },
+      },
+    });
+
+    if (payload.error) {
+      console.log('[error]', payload.error);
+      this.setState({
+        result: <ErrorResult>{payload.error.message}</ErrorResult>,
```

Comment:
Could we set `errorMessage` and `paymentMethod` fields in state instead, and then render the JSX conditionally 
in the `render` method? Setting JSX in state feels weird.